osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -419,6 +422,11 @@ private void handleDifficultyControlPoint(DifficultyControlPoint newPoint)
             if (newPoint.EquivalentTo(existing))
                 return;
 
+            // autogenerated points should not replace non-autogenerated.
+            // this allows for incorrectly ordered timing points to still be correctly handled.
+            if (newPoint.AutoGenerated && !existing.AutoGenerated)
```

Comment:
Surely this is not correct. If they're not at the same offset then the auto-generated one _should_ replace the existing one.

Reply:
This condition is also not done everywhere else `AutoGenerated` is set.

Reply:
Following on from this I've found the relevant code in stable:

```
public int CompareTo(ControlPoint other)
{
    if (Offset == other.Offset)
        return other.TimingChange.CompareTo(TimingChange);

    return Offset.CompareTo(other.Offset);
}
```

Which seems like it could directly translate for our comparer, based on the `AutoGenerated` property.

Reply:
Just to clarify, you're suggesting we drop the early exit and use a comparer instead? I'm not sure which I prefer – using the current logic means we have potentially fewer control points in the list, but it is quite an edge case...

Reply:
Should be fine as is.